### PR TITLE
executor: apply the `vecGroupChecker` in `window` executor

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve? 
Apply the `vecGroupChecker` in `window` executor

### What is changed and how it works?
Use the ```vecGroupChecker``` in ```window``` executor can make the aggregation in window executor vectorized.

benchmark for [```WindowRows```](https://github.com/pingcap/tidb/blob/master/executor/benchmark_test.go#L489)
```
name                                                 old time/op    new time/op    delta
WindowRows/(func:row_number,_ndv:10,_rows:1000)-4       179µs ±12%     109µs ±18%  -38.78%  (p=0.000 n=10+10)
WindowRows/(func:row_number,_ndv:1000,_rows:1000)-4     392µs ±15%     223µs ±21%  -43.21%  (p=0.000 n=9+10)
WindowRows/(func:row_number,_ndv:10,_rows:100000)-4    19.1ms ±52%     6.1ms ±10%  -68.28%  (p=0.000 n=10+9)
WindowRows/(func:row_number,_ndv:100000,_rows:100000)-4    26.6ms ± 7%    15.3ms ±27%  -42.63%  (p=0.000 n=10+9)

name                                                 old alloc/op   new alloc/op   delta
WindowRows/(func:row_number,_ndv:10,_rows:1000)-4      82.5kB ± 0%    82.1kB ± 0%   -0.45%  (p=0.000 n=10+10)
WindowRows/(func:row_number,_ndv:1000,_rows:1000)-4     106kB ± 1%      82kB ± 1%  -22.80%  (p=0.000 n=10+10)
WindowRows/(func:row_number,_ndv:10,_rows:100000)-4    12.1MB ± 0%    12.1MB ± 0%   -0.02%  (p=0.000 n=10+8)
WindowRows/(func:row_number,_ndv:100000,_rows:100000)-4    10.3MB ± 0%     6.2MB ± 0%  -39.23%  (p=0.000 n=10+10)

name                                                 old allocs/op  new allocs/op  delta
WindowRows/(func:row_number,_ndv:10,_rows:1000)-4         162 ± 0%       159 ± 0%   -1.85%  (p=0.000 n=10+10)
WindowRows/(func:row_number,_ndv:1000,_rows:1000)-4     1.68k ± 1%     1.07k ± 0%  -36.73%  (p=0.000 n=10+10)
WindowRows/(func:row_number,_ndv:10,_rows:100000)-4     5.17k ± 0%     5.07k ± 0%   -1.97%  (p=0.000 n=10+9)
WindowRows/(func:row_number,_ndv:100000,_rows:100000)-4      166k ± 0%      103k ± 0%  -38.07%  (p=0.000 n=10+10)
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by SunRunAway** [comment]
Could you add a benchmark result?

**Comment by Reminiscent** [comment]
> Could you add a benchmark result?

Update. PTAL @SunRunAway  @qw4990  @XuHuaiyu

**Comment by qw4990** [comment]
Please add more test cases with larger `NDV` and `Rows` like `ndv=200000 rows=1000000`.

File: executor/window.go [hunk 1]
```diff
@@ -125,11 +149,7 @@ func (e *WindowExec) consumeGroupRows(groupRows []chunk.Row) (err error) {
 	return nil
 }
 
-func (e *WindowExec) fetchChildIfNecessary(ctx context.Context) (EOF bool, err error) {
-	if e.inputIter != nil && e.inputIter.Current() != e.inputIter.End() {
-		return false, nil
-	}
-
+func (e *WindowExec) fetchChild(ctx context.Context) (EOF bool, err error) {
```
Comment (qw4990) [inline executor/window.go]: How about changing the output arguments to `(result *Chunk, EOF bool, err error)` then we can remove the `childResult` field in `WindowExec` which makes this executor simpler.

Comment (qw4990) [reply executor/window.go]: Rest LGTM

**Pull request merged**